targets/TARGET_Analog_Devices/TARGET_ADUCM302X/TARGET_ADUCM3029/bsp/rng/adi_rng.c

```diff
@@ -0,0 +1,796 @@
+#ifdef ADI_DEBUG
+    if (NULL != pDevice->pData)
+    {
+        return ADI_RNG_ALREADY_INITIALIZED;
+    }
+#endif
+
+    /* Set the internal device data */
+    pDevice->pData = pMemory;
+
+    /* initialize internal device data */
+    pDevice->pData->IRQn    = RNG0_EVT_IRQn;
+    pDevice->pData->CBFunc  = NULL;
+
+    /* clear any pending interrupts. Both bits are write 1 to clear */
+    pDevice->pRNG->STAT = BITM_RNG_STAT_RNRDY | BITM_RNG_STAT_STUCK;
+
+    /* Set the RNG register based on static configuration */
+    pDevice->pRNG->CTL = (uint16_t)RNG0_CFG_ONLY_8_BIT << BITP_RNG_CTL_SINGLE;
+    pDevice->pRNG->LEN = (RNG0_CFG_LENGTH_RELOAD << BITP_RNG_LEN_RELOAD)
+                       | (RNG0_CFG_LENGTH_PRESCALER << BITP_RNG_LEN_PRESCALE);
+
+   /* The interrupt handler only gets used in the case of callback mode so its
+    * enabling only happens in the adi_rng_RegisterCallBack API.
+    */
+    NVIC_ClearPendingIRQ(pDevice->pData->IRQn);
+
+    /* store handle at application handle pointer */
+    *phDevice = pDevice;
+
+    return ADI_RNG_SUCCESS;
+}
+
+
+/*!
+ * @brief  Uninitializes and deallocates the RNG device.
+ *
+ * @param[in]   hDevice    Device handle obtained from adi_rng_Open().
+ *
+ * @return      Status
+ *                - #ADI_RNG_SUCCESS                    Call completed successfully.
+ *                - #ADI_RNG_BAD_DEV_HANDLE [D]     Invalid device handle parameter.
+ *                - #ADI_RNG_NOT_INITIALIZED [D]    Device has not been initialized for use, see #adi_rng_Open().
+ *
+ * Uninitialize and release an allocated RNG device for other use.  The core NVIC RNG interrupt is disabled.
+ *
+ * @sa        adi_rng_Open().
+ */
+ADI_RNG_RESULT adi_rng_Close(ADI_RNG_HANDLE hDevice)
+{
+    ADI_RNG_DEV_TYPE *pDevice = (ADI_RNG_DEV_TYPE*)hDevice;
+
+#ifdef ADI_DEBUG
+    if (ADI_RNG_INVALID_HANDLE(pDevice)){
+        return ADI_RNG_BAD_DEV_HANDLE;
+    }
+
+    if (NULL == pDevice->pData) {
+        return ADI_RNG_NOT_INITIALIZED;
+    }
+#endif
+
+    /* uninitialize */
+    NVIC_DisableIRQ(pDevice->pData->IRQn);
+    pDevice->pData = NULL;
+
+    return ADI_RNG_SUCCESS;
+}
+
+/*!
+ * @brief  Enables/Disables the RNG device.
+ *
+ * @param[in]   hDevice       Device handle obtained from adi_rng_Open().
+ * @param[in]   bFlag         Flag to specify whether to enable or disable RNG device.
+ *
+ * @return      Status
+ *                - #ADI_RNG_SUCCESS                    Call completed successfully.
+ *                - #ADI_RNG_BAD_DEV_HANDLE [D]     Invalid device handle parameter.
+ *                - #ADI_RNG_NOT_INITIALIZED [D]    Device has not been initialized for use, see #adi_rng_Open().
+ *
+ * @sa        adi_rng_Open().
+ * @sa        adi_rng_RegisterCallback().
+ */
+ADI_RNG_RESULT adi_rng_Enable (ADI_RNG_HANDLE const hDevice, bool const bFlag)
+{
+    ADI_RNG_DEV_TYPE *pDevice = (ADI_RNG_DEV_TYPE*)hDevice;
+    ADI_INT_STATUS_ALLOC();
+
+#ifdef ADI_DEBUG
+    if (ADI_RNG_INVALID_HANDLE(pDevice)) {
```

Comment:
Why guard these vital checks inside debug flag?

Reply:
@dave-wu Is this also for internal testing by the drivers developers? Please check, it seems to me that this check is important not only for debug

Reply:
@RonEld Had another look and I think you are right, these should be enabled as they check for invalid handles and they are the conditions that return the prescribed error codes. I will need to discuss with the driver development guys to get them to fix this in their next release. If you are OK with it I'll leave them for now and add in the fix when the new BSP becomes available.

Reply:
AS long as you are aware of this, and that you will **have to** make a new release, I'm fine